core/src/array/array_sorted_read_state.cc

```diff
@@ -2437,30 +2088,57 @@ Status ArraySortedReadState::read_dense_sorted_col() {
     return array_->read_default(
         copy_state_.buffers_, copy_state_.buffer_sizes_);
 
-  // Iterate over each tile slab
+  // Handle resume copy
+  if (resume_copy_) {
```

Comment:
Making it static would not be cleaner because you access private members in the algorithm.  You can just pass the member functions as arguments.